### PR TITLE
logical and arithmetic right shifts

### PR DESCRIPTION
**Pull request by arasuarun**

### CODE REVIEW

File: src/jolt/instruction/sra.rs [hunk 1]
```diff
@@ -0,0 +1,84 @@
+use ark_ff::PrimeField;
+use ark_std::log2;
+
+use super::JoltInstruction;
+use crate::jolt::subtable::{
+  identity::IdentitySubtable, srl::SrlSubtable, sra_sign::SraSignSubtable, truncate_overflow::TruncateOverflowSubtable,
+  LassoSubtable,
+};
+use crate::utils::instruction_utils::{chunk_and_concatenate_for_shift, concatenate_lookups};
```
Comment (moodlezoup) [inline src/jolt/instruction/sra.rs:9]: Unused imports
```suggestion
use crate::jolt::subtable::{
  srl::SrlSubtable, sra_sign::SraSignSubtable, LassoSubtable,
};
use crate::utils::instruction_utils::{chunk_and_concatenate_for_shift};
```

File: src/jolt/subtable/srl.rs [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+#[cfg(test)]
+mod test {
+  use ark_curve25519::Fr;
+
+  use crate::{
+    jolt::subtable::{srl::SrlSubtable, LassoSubtable},
+    subtable_materialize_mle_parity_test,
+  };
+
+  subtable_materialize_mle_parity_test!(srl_materialize_mle_parity, SrlSubtable<Fr, 0>, Fr, 256);
```
Comment (moodlezoup) [inline src/jolt/subtable/srl.rs:98]: might be a good idea to test some other values of `CHUNK_INDEX`
```suggestion
  subtable_materialize_mle_parity_test!(srl_materialize_mle_parity, SrlSubtable<Fr, 0>, Fr, 256);
  subtable_materialize_mle_parity_test!(srl_materialize_mle_parity, SrlSubtable<Fr, 1>, Fr, 256);
  subtable_materialize_mle_parity_test!(srl_materialize_mle_parity, SrlSubtable<Fr, 2>, Fr, 256);
```

Comment (moodlezoup) [reply src/jolt/subtable/srl.rs:98]: same for `SllSubtable` in fact

File: src/jolt/subtable/sra_sign.rs [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+    let operand_chunk_width: usize = (log2(M) / 2) as usize;
+
+    // find position of sign bit in the chunk 
+    let sign_bit_index = 63 % operand_chunk_width; 
+    let sign_bit_mask: usize = 1 << (sign_bit_index);
+
+    for idx in 0..M {
+      let (x, y) = split_bits(idx, operand_chunk_width);
+
+      let x_sign = F::from(((x & sign_bit_mask) >> sign_bit_index) as u64);
```
Comment (moodlezoup) [inline src/jolt/subtable/sra_sign.rs:36]: slightly cleaner imo
```suggestion
      let x_sign = F::from(((x >> sign_bit_index) & 1) as u64);
```

File: src/jolt/subtable/sra_sign.rs [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+    }
+    entries
+  }
+
+  fn evaluate_mle(&self, point: &[F]) -> F {
+    // first half is chunk X_i
+    // and second half is always chunk Y_0
+    debug_assert!(point.len() % 2 == 0);
+
+    let MAX_SHIFT: usize = 64;
```
Comment (moodlezoup) [inline src/jolt/subtable/sra_sign.rs:51]:
```suggestion
    const MAX_SHIFT: usize = 64;
```

File: src/jolt/subtable/srl.rs [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+    }
+    entries
+  }
+
+  fn evaluate_mle(&self, point: &[F]) -> F {
+    // first half is chunk X_i
+    // and second half is always chunk Y_0
+    debug_assert!(point.len() % 2 == 0);
+
+    let MAX_SHIFT = 64;
```
Comment (moodlezoup) [inline src/jolt/subtable/srl.rs:49]:
```suggestion
    const MAX_SHIFT: usize = 64;
```

**Review by moodlezoup — COMMENTED** [review]
looking good, just a few minor suggestions

**Comment by arasuarun** [comment]
Done. Made the changes you suggested and removed unused imports from many instruction and subtable files.

**Review by moodlezoup — APPROVED** [review]
thanks!

**Pull request merged**